src/cls/rgw/cls_rgw_client.cc

```diff
@@ -834,16 +834,10 @@ int cls_rgw_usage_log_trim(IoCtx& io_ctx, const string& oid, const string& user,
   call.bucket = bucket;
   encode(call, in);
 
-  bool done = false;
-  do {
-    ObjectWriteOperation op;
-    op.exec(RGW_CLASS, RGW_USER_USAGE_LOG_TRIM, in);
-    int r = io_ctx.operate(oid, &op);
-    if (r == -ENODATA)
-      done = true;
-    else if (r < 0)
-      return r;
-  } while (!done);
+  ObjectWriteOperation op;
+  op.exec(RGW_CLASS, RGW_USER_USAGE_LOG_TRIM, in);
+  int r = io_ctx.operate(oid, &op);
+  if (r < 0 && r == -ENODATA) return r;
```

Comment:
if i understand this correctly, it's moving the entire unbounded loop down to cls_rgw? we can't loop indefinitely in a single cls call or we'll trigger osd suicide timeouts - that's the main purpose of MAX_USAGE_TRIM_ENTRIES

is there a simpler version of this fix that preserves this limit of MAX_USAGE_TRIM_ENTRIES per osd op?